crates/modelardb_server/src/remote.rs

```diff
@@ -669,6 +752,37 @@ impl FlightService for FlightServiceHandler {
                     body: batch_bytes.into(),
                 })
             }))))
+        } else if action.r#type == "UpdateRemoteObjectStore" {
+            // If the type of the new remote object store is not "s3" or "azureblobstorage", return an error.
+            let (object_store_type, offset_data) = extract_argument(&action.body)?;
+
+            let object_store = match object_store_type {
+                "s3" => parse_s3_arguments(offset_data).await,
+                "azureblobstorage" => parse_azure_blob_storage_arguments(offset_data).await,
+                _ => Err(Status::unimplemented(format!(
+                    "{object_store_type} is currently not supported."
+                ))),
+            }?;
+
+            // Update the object store used for data transfers.
+            let mut storage_engine = self.context.storage_engine.write().await;
+            storage_engine
+                .update_remote_data_folder(object_store)
+                .await
+                .map_err(|error| {
+                    Status::internal(format!("Could not update remote data folder: {error}"))
+                })?;
+
+            // If on a cloud node, both the remote data folder and the query data folder should be updated.
+            if self.context.metadata_manager.node_type == NodeType::Cloud {
```

Comment:
Maybe move this to line 756 and immediately return an error if this is a cloud node, otherwise it becomes possible to have a different remote object store for the data transfer component and the query engine.

Reply:
Fixed.